intel_extension_for_transformers/llm/runtime/graph/models/gptneox/gptneox_utils.cpp

```diff
@@ -193,3 +181,113 @@ class gptneox_quant_layer : public quant_layer_base {
   }
 };
 REGISTER_QUANT_LAYER_CLASS(gptneox);
+
+class gptneox_beam_search_kv_cache_reorder : public beam_search_kv_cache_reorder {
+ public:
+  explicit gptneox_beam_search_kv_cache_reorder(model_context* lctx) : beam_search_kv_cache_reorder(lctx) {}
+  ~gptneox_beam_search_kv_cache_reorder() {}
+
+  virtual void update(const uint32_t& n_past, const uint32_t& n_prompt_tokens,
+                      const std::vector<std::tuple<int, int>>& kv_reorder_indices = {},
+                      const std::vector<beam>& next_beams = {}) override {
+    // first step
+    if (n_past == n_prompt_tokens) {
```

Comment:
what's the difference between this and its base class https://github.com/intel/intel-extension-for-transformers/blob/4f5de037f524de40839ccc524664f98f2937d711/intel_extension_for_transformers/llm/runtime/graph/models/model_utils/model_utils.cpp#L2085

Reply:
`k` layout is `[head_dim, N, n_head] ` in `gpt-neox` rather `[n_embd, N] ` in `gpt-j`